mm2src/coins/z_coin/z_coin_errors.rs

```diff
@@ -1,258 +1,311 @@
-            | GenTxError::BlockchainScanStopped
-            | GenTxError::LightClientErr(_)
-            | GenTxError::FailedToCreateNote => WithdrawError::InternalError(gen_tx.to_string()),
+                },
+                GenTxError::Rpc(e) => WithdrawError::Transport(e.to_string()),
+                GenTxError::DecryptedOutputNotFound
+                | GenTxError::FailedToGetMerklePath
+                | GenTxError::PrevTxNotConfirmed
+                | GenTxError::GetWitnessErr(_)
+                | GenTxError::NumConversion(_)
+                | GenTxError::TxBuilderError(_)
+                | GenTxError::TxReadError { .. }
+                | GenTxError::BlockchainScanStopped
+                | GenTxError::LightClientErr(_)
+                | GenTxError::SpendableNotesError(_)
+                | GenTxError::FailedToCreateNote => WithdrawError::InternalError(gen_tx.to_string()),
+            }
         }
     }
-}
 
-impl From<BlockchainScanStopped> for GenTxError {
-    #[inline]
-    fn from(_: BlockchainScanStopped) -> Self { GenTxError::BlockchainScanStopped }
-}
+    impl From<BlockchainScanStopped> for GenTxError {
+        #[inline]
+        fn from(_: BlockchainScanStopped) -> Self { GenTxError::BlockchainScanStopped }
+    }
 
-#[derive(Debug, Display)]
-#[allow(clippy::large_enum_variant)]
-pub enum SendOutputsErr {
-    GenTxError(GenTxError),
-    NumConversion(NumConversError),
-    Rpc(UtxoRpcError),
-    TxNotMined(String),
-    PrivKeyPolicyNotAllowed(PrivKeyPolicyNotAllowed),
-}
+    #[derive(Debug, Display)]
+    #[allow(clippy::large_enum_variant)]
+    pub enum SendOutputsErr {
+        GenTxError(GenTxError),
+        NumConversion(NumConversError),
+        Rpc(UtxoRpcError),
+        TxNotMined(String),
+        PrivKeyPolicyNotAllowed(PrivKeyPolicyNotAllowed),
+    }
 
-impl From<PrivKeyPolicyNotAllowed> for SendOutputsErr {
-    fn from(err: PrivKeyPolicyNotAllowed) -> Self { SendOutputsErr::PrivKeyPolicyNotAllowed(err) }
-}
+    impl From<PrivKeyPolicyNotAllowed> for SendOutputsErr {
+        fn from(err: PrivKeyPolicyNotAllowed) -> Self { SendOutputsErr::PrivKeyPolicyNotAllowed(err) }
+    }
 
-impl From<GenTxError> for SendOutputsErr {
-    fn from(err: GenTxError) -> SendOutputsErr { SendOutputsErr::GenTxError(err) }
-}
+    #[derive(Debug, Display)]
+    pub enum GetUnspentWitnessErr {
+        EmptyDbResult,
+        TreeOrWitnessAppendFailed,
+        OutputCmuNotFoundInCache,
+        Sql(SqliteError),
+    }
 
-impl From<NumConversError> for SendOutputsErr {
-    fn from(err: NumConversError) -> SendOutputsErr { SendOutputsErr::NumConversion(err) }
-}
+    impl From<SqliteError> for GetUnspentWitnessErr {
+        fn from(err: SqliteError) -> GetUnspentWitnessErr { GetUnspentWitnessErr::Sql(err) }
+    }
 
-impl From<UtxoRpcError> for SendOutputsErr {
-    fn from(err: UtxoRpcError) -> SendOutputsErr { SendOutputsErr::Rpc(err) }
-}
+    #[derive(Debug, Display)]
+    pub enum ZCoinBuildError {
```

Comment:
`ZCoinBuildError` is duplicated for wasm, can't both be unified? Please try to do that in this PR.

I am against duplicating structs/functions for wasm in general, it makes maintaining this code in the future difficult, that's why a common implementation for all of those is prefered. I will allow this for some of the functions in this PR since it's just a preparation PR, but at the end all of these duplications have to be unified.